src/libasr/codegen/asr_to_c_cpp.h

```diff
@@ -1077,15 +1084,6 @@ PyMODINIT_FUNC PyInit_lpython_module_)" + fn_name + R"((void) {
                     || param->m_intent == ASRUtils::intent_out)
                     && !ASRUtils::is_aggregate_type(param->m_type))) {
                     args += "&" + src;
-                } else if (param->m_intent == ASRUtils::intent_out) {
-                    if (ASR::is_a<ASR::List_t>(*param->m_type)) {
-                        ASR::List_t* list_type = ASR::down_cast<ASR::List_t>(param->m_type);
-                        if (list_type->m_type->type == ASR::ttypeType::CPtr){
-                            args += "&" + src;
-                        }
-                    } else {
-                        args += src;
-                    }
```

Comment:
Should be reverted as per https://github.com/lcompilers/lpython/commit/f9ab31ebc99f6a76f9d3d1e162e37e667e550d05